src/connection.c

```diff
@@ -45,6 +45,8 @@ static noPollConnOpts * createConnOpts (char * extra_headers, bool secure);
 static noPollConn * nopoll_tls_common_conn (noPollCtx  * ctx,char * serverAddr,char *serverPort,char * extra_headers);
 static char* build_extra_headers( const char *auth, const char *device_id,
                                   const char *user_agent, const char *convey );
+//Flag for Connection Fallback
+int FLAGS_FALLBACK_CONN = 0;
```

Comment:
Instead of this global FLAGS_FALLBACK_CONN flag being shared better to use new config flag to indicate fallback 
#define FLAGS_IPV6_IPV4 (FLAGS_IPV6_ONLY | FLAGS_IPV4_ONLY) i.e. 0x03
Use a tmpFlag func local variable in createNopollConnection() to be either FLAGS_IPV6_ONLY or FLAGS_IPV4_ONLY during each retry and pass this to nopoll_tls_common_conn() func as an argument.